crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -198,11 +224,17 @@ impl PartitionProducer {
 
         let mut batch_notifiers = vec![];
 
+        let mut flush_total_bytes = 0;
+
         for p_batch in batches_ready {
             let notify = p_batch.notify.clone();
             let batch = p_batch.batch();
 
-            let raw_batch = batch.try_into()?;
+            let raw_batch: Batch<RawRecords> = batch.try_into()?;
+
+            if self.client_stats.is_some() {
+                flush_total_bytes += BATCH_FILE_HEADER_SIZE + raw_batch.records().0.len();
+            }
```

Comment:
It would be more clear like this:
```
if let Some(stat) = self.client_stat. {
   state.update_flush_bytes(raw.batch_len());
}
```

Reply:
This is fixed. There were bugs in the batch type conversions where we were losing the batch length.